qutip/solver/steadystate.py

```diff
@@ -268,6 +278,28 @@ def _steadystate_svd(L, **kw):
     return rho / rho.tr()
 
 
+def _steadystate_expm(L, rho=None, propagator_tol=1e-5, **kw):
+    if rho is None:
+        from qutip import rand_dm
+        rho = rand_dm(L.dims[0][0])
+    # Propagator at an arbitrary long time
+    prop = (100 * L).expm()
+
+    rho_prev = qzero_like(rho)
+    niter = 0
+    while hilbert_dist(rho, rho_prev) > propagator_tol and niter < 30:
+        rho_prev = rho
+        rho = prop(rho)
+        prop = prop @ prop
+        rho = (rho + rho.dag()) / (2 * rho.tr())
+        niter += 1
```

Comment:
```suggestion
    dist = 1.0 + propagator_rol
    niter = 0
    while niter < 30:
        rho_next = prop(rho)
        rho_next = (rho_next + rho_next.dag()) / (2 * rho_next.tr())
        if hilbert_dist(rho_next, rho) <= propagator_tol:
            return rho_next
        rho = rho_next
        prop = prop @ prop
        niter += 1
```
While the original is terse it calculates a `prop` that is ununsed and starts the first iteration with a `hilbert_dist(...)` comparison that always fails. There are probably further small improvements that can be made to the above.